go-controller/pkg/ovn/ovn.go

```diff
@@ -814,10 +814,10 @@ func (oc *Controller) WatchEgressIP() {
 			if err := oc.addEgressIP(eIP); err != nil {
 				klog.Error(err)
 			}
-			oc.eIPC.assignmentRetryMutex.Unlock()
 			if err := oc.updateEgressIPWithRetry(eIP); err != nil {
 				klog.Error(err)
 			}
+			oc.eIPC.assignmentRetryMutex.Unlock()
```

Comment:
use defer instead right after you lock, since this is the end of the function. Same for update function.